torch/nn/modules/module.py

```diff
@@ -551,6 +551,81 @@ def __repr__(self):
         tmpstr = tmpstr + ')'
         return tmpstr
 
+    def summary(self, input_size):
+        def register_hook(module):
+            def hook(module, input, output):
+                class_name = str(module.__class__).split('.')[-1].split("'")[0]
+                module_idx = len(summary)
+                m_key = '%s-%i' % (class_name, module_idx + 1)
+                summary[m_key] = OrderedDict()
+                summary[m_key]['input_shape'] = list(input[0].size())
```

Comment:
I think `input` can be a tuple of Variables, right?

Reply:
I could be missing something, but if `input` is a tuple of Variables I can still extract the module input shape with `input[0].size()`, no?